### PR TITLE
Show Earth's mass and show all semi-axes for non spherical bodies

### PR DESCRIPTION
**Pull request by JiliTheSpaceboy**
1. Make it so Earth's mass is shown, considering that Jupiter's mass is also shown even if it simply appears as 1.000 MJ

2. Objects that are neither spherical or use mesh now show mean radius along with all three semi-axes, using the following format:
Radius: mean (a × b × c)

3. kg masses can now uses scientific notation if sufficiently large. No more 20,000,000,000,000,000,000 kg, just 2E+19 kg.

### CODE REVIEW

**Comment by JiliTheSpaceboy** [comment]
I don't know how to compile so this is untested and likely needs some fixing, so I decided to make a pull request.

**Comment by JiliTheSpaceboy** [comment]
TODO:
- Make small masses (unit: kg) use scientific notation
- Move radius if elses into displayRadius function
- Clean up Radius display; for triaxial radii, make all values use the same unit and putting said unit only once at the end; possible dynamic sigfig?

File: src/celestia/hud.cpp [hunk 1]
```diff
@@ -184,10 +184,10 @@ displayMass(const util::NumberFormatter& formatter, Overlay& overlay, float mass
     if (mass < 0.001f)
     {
         if (measurement == MeasurementSystem::Imperial)
-            overlay.print(_("Mass: {} lb\n"),
+            overlay.print(_("Mass: {%.4g} lb\n"),
                           formatter.format(mass * astro::EarthMass / static_cast<float>(OneLbInKg), 4, SigDigitNum));
         else
-            overlay.print(_("Mass: {} kg\n"),
+            overlay.print(_("Mass: {%.4g} kg\n"),
```
Comment (ajtribick) [inline src/celestia/hud.cpp:190]: also needs to updates in formatnum.*

**Review by ajtribick — CHANGES_REQUESTED** [review]

File: src/celestia/hud.cpp [hunk 2]
```diff
@@ -184,10 +184,10 @@ displayMass(const util::NumberFormatter& formatter, Overlay& overlay, float mass
     if (mass < 0.001f)
     {
         if (measurement == MeasurementSystem::Imperial)
-            overlay.print(_("Mass: {} lb\n"),
+            overlay.print(_("Mass: {%.4g} lb\n"),
```
[the remark this answers is not in the document]
Comment (JiliTheSpaceboy) [reply src/celestia/hud.cpp:187]: Should I just switch this back for now? Unless a proper developer is willing to work on this (maybe as part of your current plan to upgrade the Radius display?), I'm not sure I can handle this one.

[the remark this answers is not in the document]
Comment (ajtribick) [reply src/celestia/hud.cpp:187]: Yep, I think the other changes in this to add the semiaxes and fix the missing mass are pretty useful, so let's keep those

[the remark this answers is not in the document]
Comment (JiliTheSpaceboy) [reply src/celestia/hud.cpp]: I've reverted that part.

**Review by ajtribick — APPROVED** [review]

File: src/celestia/hud.cpp [hunk 3]
```diff
@@ -533,13 +533,28 @@ displayPlanetInfo(const util::NumberFormatter& formatter,
...
     if (body.getClassification() == BodyClassification::Invisible)
     {
         return;
     }
-
-    overlay.printf(_("Radius: %s\n"),
-                   DistanceKmToStr(formatter, body.getRadius(), 5, hudSettings.measurementSystem));
+    else if (body.isEllipsoid() && !body.isSphere()) // show mean radius along with triaxial semi-axes
+    {
+        double radiusMean = cbrt(semiAxes.prod());
```
Comment (375gnu) [inline src/celestia/hud.cpp:543]:
```suggestion
        Eigen::Vector3f semiAxes = body.getSemiAxes();
        double radiusMean = cbrt(semiAxes.prod());
```

File: src/celestia/hud.cpp [hunk 4]
```diff
@@ -533,13 +533,28 @@ displayPlanetInfo(const util::NumberFormatter& formatter,
     overlay.printf(_("Distance: %s\n"),
                    DistanceKmToStr(formatter, distance, 5, hudSettings.measurementSystem));
 
+    Eigen::Vector3f semiAxes = body.getSemiAxes();
```
Comment (375gnu) [inline src/celestia/hud.cpp:536]:
```suggestion
```

**Pull request merged**